tests/test_compdat.py

```diff
@@ -500,3 +512,103 @@ def test_msw_str2df():
+
+    # Test WSEGVALV
+    assert len(wsegvalv) == 1
+    assert "WELL" in wsegvalv
+    assert wsegvalv["WELL"].unique()[0] == "OP_6"
+    assert len(wsegvalv.dropna(axis=1, how="all").iloc[0]) == 5
+
+    schstr_valv = """
+WELSPECS
+   'OP_6' 'DUMMY' 28 37 1575.82 OIL 0.0 'STD' 'SHUT' 'YES' 0 'SEG' /
+/
+
+COMPDAT
+    'OP_6' 28 37 1 1 OPEN 0 1.2719 0.311 114.887 0.0 0.0 'X' 19.65 /
+/
+
+WELSEGS
+-- WELL   SEGMENTTVD  SEGMENTMD WBVOLUME INFOTYPE PDROPCOMP MPMODEL
+   'OP_6'        0.0        0.0   1.0E-5    'ABS'     'HF-'    'HO' /
+--  SEG  SEG2  BRANCH  OUT MD       TVD       DIAM ROUGHNESS
+     32    32    1       31  2371.596 1577.726  0.15 0.00065    /
+/
+
+COMPSEGS
+   'OP_6' /
+--  I   J   K   BRANCH STARTMD  ENDMD    DIR DEF  SEG
+    28  37   1   2     2366.541 2376.651  1*  3*  32   /
+/
+
+-- Max defaulted
+WSEGVALV
+-- WELL   SEG             CV      AC
+    OP_6  31       0.0084252 0.00075 /
+/
+"""
+
+    # Test WSEGVALV max defaulted
+    deck = EclFiles.str2deck(schstr_valv)
+    compdfs = compdat.deck2dfs(deck)
+    wsegvalv = compdfs["WSEGVALV"]
+    assert len(wsegvalv) == 1
+    assert "WELL" in wsegvalv
+    assert wsegvalv["WELL"].unique()[0] == "OP_6"
+    assert len(wsegvalv.dropna(axis=1, how="all").iloc[0]) == 5
+
+    schstr_valv_wlist = """
+WELSPECS
+   'OP_6' 'DUMMY' 28 37 1575.82 OIL 0.0 'STD' 'SHUT' 'YES' 0 'SEG' /
+   'OP_7' 'DUMMY' 51 15 1675.82 OIL 0.0 'STD' 'SHUT' 'YES' 0 'SEG' /
+/
+
+COMPDAT
+    'OP_6' 28 37 1 1 OPEN 0 1.2719 0.311 114.887 0.0 0.0 'X' 19.65 /
+    'OP_7' 51 15 1 1 OPEN 0 1.2719 0.311 114.887 0.0 0.0 'X' 19.65 /
+/
+
+WELSEGS
+-- WELL   SEGMENTTVD  SEGMENTMD WBVOLUME INFOTYPE PDROPCOMP MPMODEL
+   'OP_6'      1575.82      0.0   1.0E-5    'ABS'     'HF-'    'HO' /
+--  SEG  SEG2  BRANCH  OUT MD       TVD       DIAM ROUGHNESS
+     2    2     1        1  2371.596 1577.726  0.15 0.00065    /
+/
+
+WELSEGS
+-- WELL   SEGMENTTVD  SEGMENTMD WBVOLUME INFOTYPE PDROPCOMP MPMODEL
+   'OP_7'      1575.82      0.0   1.0E-5    'ABS'     'HF-'    'HO' /
+--  SEG  SEG2  BRANCH  OUT MD       TVD       DIAM ROUGHNESS
+     2    2     1        1  2071.596 1577.726  0.15 0.00065    /
+/
+
+COMPSEGS
+   'OP_6' /
+--  I   J   K   BRANCH STARTMD  ENDMD    DIR DEF  SEG
+    28  37   1   2     2366.541 2376.651  1*  3*   2   /
+/
+COMPSEGS
+   'OP_7' /
+--  I   J   K   BRANCH STARTMD  ENDMD    DIR DEF  SEG
+    51  15   1   2     2080.541 2099.651  1*  3*    2   /
+/
+
+WLIST
+-- NAME  OPERATION      WELLS
+ OPRODS        NEW  OP_6 OP_7 /
+/
+
+-- Max defaulted with well list
+WSEGVALV
+--   LIST SEG              CV      AC
+   OPRODS  31       0.0084252 0.00075 /
+/
+"""
+
+    # Test WSEGVALV max defaulted and wlist
+    deck = EclFiles.str2deck(schstr_valv_wlist)
+    compdfs = compdat.deck2dfs(deck)
+    wsegvalv = compdfs["WSEGVALV"]
+    assert len(wsegvalv) == 1
+    assert "WELL" in wsegvalv
+    assert wsegvalv["WELL"].unique()[0] != "OP_6"
```

Comment:
Add some sample tests towards the numerical values in the dataframe, checking that CV column actually has a specific value at the correct row perhaps?